backend/protos/xyz/block/ftl/v1/ftl.proto

```diff
@@ -375,3 +375,129 @@ service RunnerService {
   // Terminate the deployment on this Runner.
   rpc Terminate(TerminateRequest) returns (RegisterRunnerRequest);
 }
+
+message ConfigRef {
+  optional string module = 1;
+  string name = 2;
+}
+
+enum ConfigProvider {
+  // Write values inline in the configuration file.
+  CONFIG_INLINE = 0;
+
+  // Print configuration as environment variables.
+  CONFIG_ENVAR = 1;
+}
+
+message ListConfigRequest {
+  optional ConfigProvider provider = 1;
+  optional string module = 2;
+  optional bool include_values = 3;
+}
+message ListConfigResponse {
+  message Config {
+    string refPath = 1;
+    optional string value = 2;
+  }
+  repeated Config configs = 1;
+}
+
+message GetConfigRequest {
+  optional ConfigProvider provider = 1;
+  ConfigRef ref = 2;
+}
+message GetConfigResponse {
+  string value = 1;
+}
+
+message SetConfigRequest {
+  optional ConfigProvider provider = 1;
+  ConfigRef ref = 2;
+  string value = 3;
+}
+message SetConfigResponse {}
+
+message UnsetConfigRequest {
+  optional ConfigProvider provider = 1;
+  ConfigRef ref = 2;
+}
+message UnsetConfigResponse {}
+
+enum SecretProvider {
+  // Write values inline in the configuration file.
+  SECRET_INLINE = 0;
+
+  // Print configuration as environment variables.
+  SECRET_ENVAR = 1;
+
+  // Write to the system keychain.
+  SECRET_KEYCHAIN = 2;
+
+  // Store a secret in the 1Password vault.
+  SECRET_OP = 3;
+}
+
+message ListSecretsRequest {
+  optional SecretProvider provider = 1;
+  optional string module = 2;
+  optional bool include_values = 3;
+}
+message ListSecretsResponse {
+  message Secret {
+    string refPath = 1;
+    optional string value = 2;
+  }
+  repeated Secret secrets = 1;
+}
+
+message GetSecretRequest {
+  optional SecretProvider provider = 1;
+  ConfigRef ref = 2;
+}
+message GetSecretResponse {
+  string value = 1;
+}
+
+message SetSecretRequest {
+  optional SecretProvider provider = 1;
+  ConfigRef ref = 2;
+  string value = 3;
+}
+message SetSecretResponse {}
+
+message UnsetSecretRequest {
+  optional SecretProvider provider = 1;
+  ConfigRef ref = 2;
+}
+message UnsetSecretResponse {}
+
+// AdminService centralizes project configuration access and provides CRUD operations.
+service AdminService {
+  rpc Ping(PingRequest) returns (PingResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+
+  // List configuration.
+  rpc ListConfig(ListConfigRequest) returns (ListConfigResponse);
+
+  // Get a config value.
+  rpc GetConfig(GetConfigRequest) returns (GetConfigResponse);
```

Comment:
It reads a little weird, but I think it's preferable to start the method names with the type they're operating on: `ConfigGet`, `ConfigList`, etc. It makes completion and discovery much more obvious.